Count unique listeners for Icecast

danielpilot opened the pull request:
AzuraCast is showing 0 current unique listeners when connecting to an IceCast relay

Reviewing the code, I found out, connecting to my server:

1. Method getClients was returning an empty array because $xml->source->listeners didn't exist, but $xml->source->Listeners did.
2. There is no creation for unique listeners in Icecast. Using the method getClient and the parameter unique, we can count the number of unique clients connected

danielpilot committed:
Count unique listeners for Icecast

BusterNeece commented:
@danielpilot Currently, the calculations of unique listeners and appending them to the per-mount-point report is done inside AzuraCast instead of inside this library. That's why you don't see some of the code that would be responsible for that here.

I'm not opposed to the idea of actually moving that functionality over here, because right now the NowPlaying library doesn't really match how we use it internal to AzuraCast, and likely isn't all that useful to others in its current form either, for the same reason.

As for the issue with `xml->source->listeners` vs `->Listeners`, I believe this is actually a minor version difference between builds of Icecast, since our Icecast-KH builds have the lower-case `listeners` property. We can check for either and handle either case for broader compatibility.

danielpilot commented:
@SlvrEagle23

Hello!

Ok, I didn't know the calculations for unique listeners in Icecast were done in AzuraCast and not in this library, I was looking for a solution because I just moved a relay from ShoutCast to IceCast and it's showing me 0 unique listeners all the time, even if the number of listeners is higher. I found this repository and thought that maybe it was missing from here.

And ok for the listeners case, I just tested it with my Icecast relay and that's why I saw the difference, the getClients method was returning me an empty array so I just fixed it for me to work with my case.

BusterNeece commented:
@danielpilot No worries, in your case it makes sense. I'll be updating this library shortly to handle a lot of what AzuraCast is currently doing internally, so that it can serve largely standalone and will be more cleanly integrated into AC itself.
